Your_first_ASDF_converter.ipynb

```diff
@@ -0,0 +1,554 @@
+    "And finally, the entry point reference in setup.cfg, provided here as a whole file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "feb206d7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile setup.cfg\n",
+    "[metadata]\n",
+    "name = mfconverter\n",
+    "description = TODO\n",
+    "long_description = TODO\n",
+    "author = TODO\n",
+    "version='0.1.0'\n",
+    "license = BSD-3-Clause\n",
+    "\n",
+    "[options]\n",
+    "zip_safe = True\n",
+    "python_requires = >=3.6\n",
+    "setup_requires =\n",
+    "    setuptools_scm\n",
+    "install_requires =\n",
+    "    jsonschema>=3.0.2\n",
+    "    asdf>=2.8\n",
+    "    psutil>=5.7.2\n",
+    "    numpy>=1.16\n",
+    "package_dir =\n",
+    "    =src\n",
+    "packages = find:\n",
+    "\n",
+    "[options.entry_points]\n",
+    "asdf.extensions =\n",
+    "    mfc_extension = mfconverter.extensions:get_extensions\n",
+    "\n",
+    "[options.packages.find]\n",
+    "where = src\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8d0e3882",
+   "metadata": {},
+   "source": [
+    "If you wish to learn more about entry points, see:\n",
+    "https://packaging.python.org/guides/creating-and-discovering-plugins/\n",
+    "\n",
+    "We need the setup.py file too"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3ffb22d6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile setup.py\n",
+    "#!/usr/bin/env python3\n",
+    "from setuptools import setup\n",
+    "\n",
+    "setup(use_scm_version=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b7628aaa",
+   "metadata": {},
+   "source": [
+    "Install package\n",
+    "------------------\n",
+    "\n",
+    "This is best done from a terminal window using the command\n",
+    "\n",
+    "`pip install --editable .`\n",
+    "\n",
+    "But this notebook will do that, but a consequence is that if the reinstallation must be done, the Jupyter kernel must be restarted to pick up the new installation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8e3c3246",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install --editable ."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d6aa6fb1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import asdf\n",
+    "af = asdf.AsdfFile()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "259142fa",
+   "metadata": {},
+   "source": [
+    "Before trying a full test, it is good to see if asdf knows about this converter.\n",
+    "Since the first thing that one will do is create an ASDF file with this object,\n",
+    "does ASDF know about this type?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6b4cebd9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "af.extension_manager._converters_by_type"
```

Comment:
When I got to this point, the dict was empty.  After restarting the notebook kernel my converter appeared.

Reply:
I have inserted the code to restart the kernel from within the notebook cell